radix-engine/src/system/system_modules/costing/fee_reserve.rs

```diff
@@ -534,6 +571,15 @@ impl ExecutionFeeReserve for SystemLoanFeeReserve {
         // Move resource
         self.locked_fees
             .push((vault_id, fee.take_all(), contingent));
+
+        if self.abort_when_loan_repaid && self.is_deferred_costs_and_loan_covered().unwrap_or(false)
```

Comment:
Minor..
Since `is_deferred_costs_and_loan_covered()` returns `FeeReserveError` then maybe no need to map error returned by `is_deferred_costs_and_loan_covered()` to 
`FeeReserveError::Abort(..)`?

Reply:
Good point!